### PR TITLE
Upgrading go versions

### PR DESCRIPTION
**Pull request by jillianwilson**
Updating go.mod and dockerfile to user version 1.20 of Go

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/1password/kubernetes-secrets-injector
 
-go 1.19
+go 1.20
```
Comment (volodymyrZotov) [inline go.mod:3]: Looks like vendor updates are missed after bumping a version.

Comment (jillianwilson) [reply go.mod:3]: Running go mod vendor didn't turn up any updates 🤷

Comment (edif2008) [reply go.mod:3]: That is because no go mod packages were updated.

Speaking of which, should we do this in a separate PR? 🤔

Comment (jillianwilson) [reply go.mod:3]: As a separate work effort we should update our other dependencies

Comment (volodymyrZotov) [reply go.mod:3]: Ok, sounds good!

**Review by volodymyrZotov — APPROVED** [review]

**Pull request merged**